File: recovar/gui_v2/backend/ws_types.py
"""WebSocket message type definitions.

Keep in sync with frontend/src/lib/ws_types.ts.
These are intentionally simple — no codegen overhead needed.
"""

from __future__ import annotations

from typing import Any


# Message types sent server → client
LOG_LINE = "log_line"
STATUS_CHANGE = "status_change"
PROGRESS = "progress"
RECONNECT_SYNC = "reconnect_sync"
PING = "ping"

# Message types sent client → server
PONG = "pong"

# All valid message types
ALL_TYPES = {LOG_LINE, STATUS_CHANGE, PROGRESS, RECONNECT_SYNC, PING, PONG}

File: recovar/gui_v2/frontend/src/lib/ws_types.ts
/**
 * WebSocket message types — keep in sync with backend/ws_types.py.
 */

// Server → Client
export const LOG_LINE = "log_line" as const;
export const STATUS_CHANGE = "status_change" as const;
export const PROGRESS = "progress" as const;
export const RECONNECT_SYNC = "reconnect_sync" as const;
export const PING = "ping" as const;

// Client → Server
export const PONG = "pong" as const;

export interface WsMessage<T = unknown> {
  type: string;
  data: T;
  ts: number;
}

export interface LogLineData {
  line: string;
  offset: number;
}

export interface StatusChangeData {
  old: string;
  new: string;
  error?: string;
}

export interface ProgressData {
  step: number;
  total: number;
  label: string;
}

export interface ReconnectSyncData {
  status: string;
  log_offset: number;
  log_tail: string[];
}
